default.nix

```diff
@@ -604,9 +605,9 @@ let overrideCabal = pkg: f: if pkg == null then null else lib.overrideCabal pkg
 in let this = rec {
   overrideForGhcjs = haskellPackages: haskellPackages.override {
     overrides = self: super: {
-      mkDerivation = drv: super.mkDerivation.override {
-        hscolour = ghc.hscolour;
-      } (drv // {
+      hscolour = ghc.hscolour;
```

Comment:
Wouldn't the previous behavior also unilaterally override `hscolour` regardless if it were a library or not? In the previous code, anything using `mkDerivation` would get GHC's version of `hscolour`, no? I don't know of any Haskell packages that would be built another way...unless they further overrode to change this.